### ISSUE TITLE
[Cache] Feature request to be able to have the cache longer

### ISSUE THREAD

**Issue by hilmanski** [serpapi/public-roadmap#1426]
A customer is interested if he can have a longer cache duration and be able to adjust the timing for more than 1H.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000059468?view=List)

**Comment by schaferyan** [comment]
Another user reached out requesting to have the option to change the length of the cache, this customer would like it to be shorter:

[Front](https://app.frontapp.com/open/msg_2erhl3z0?key=hOOKCR4_4wjhTkCjfnEgAWdvP3wxb0oe)

**Comment by NateSkiles** [comment]
Another user reached out requesting the ability to shorten the cache period. Their comment below:
> One easy way to do this would be allow a "custom_salt" field which was set by the user. If they wanted to not use the cache, they'd just change that field.

[Front](https://app.frontapp.com/open/msg_2euouz8c?key=oZTi4OpJc2Xm7m_8IjmW7GLKwMKfICfQ)

**Comment by hartator** [comment]
I don't fully get the goal of this.

Aren't users able to get any search via the search ID? There is no need to play with a salt or the duration of the cache. The search is there for next 30 days. And all searches are fresh if they are not exactly the same by exactly the same user. `no_cache=true` should never be used only for grabbing live news via the exact same query every 5 minutes or something in the like.

Or if they want no trace at all, maybe they mean this: #981.

**Comment by hilmanski** [comment]
I'm not sure about the second case: control the duration for less than an hour since `no_cache=true` seems like a good option.

For the original case (a request to make the cache duration longer), I believe it's more convenient for users to change only a parameter when using our API rather than storing a search ID every time. I can see the use case when the app is customer-facing; they might want to cache every search longer to avoid using credits. While it's possible to store the cache themselves or use the search ID, changing a parameter is definitely easier for them in this case.

**Comment by hartator** [comment]
> For the original case (a request to make the cache duration longer), I believe it's more convenient for users to change only a parameter when using our API rather than storing a search ID every time. I can see the use case when the app is customer-facing; they might want to cache every search longer to avoid using credits. While it's possible to store the cache themselves or use the search ID, changing a parameter is definitely easier for them in this case.

Yes, that's a good point and a valid use-case. Like Google Sheet comes in mind when a 1-day cache would be better than a 1-hour cache.

Can you try to find actual users or customers wanting this before we start working on it?

**Comment by alexbarron** [comment]
Another request:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215468829060701?view=List)

**Comment by NateSkiles** [comment]
An Enterprise customer requested this:

[Front](https://app.frontapp.com/open/cnv_1gw3m0v0?key=WWzzPCZi2kmTd1dDS1gAipp9KmC_8fN4)

Note: While extending the cache duration is helpful, this customer specifically requested the ability to _shorten_ the cache time as well. This is achievable through a custom solution today, but would be a valuable built-in feature.

Unfreezing this now, as there appears to be sufficient customer interest to begin moving it forward.

**Comment by schaferyan** [comment]
Another customer requested this:

[Front](https://app.frontapp.com/open/cnv_1i872krg?key=iQvBSg4jAdlCAP5yhV3DDGjaex6Oww5E)

**Comment by hilmanski** [comment]
A user is interested in extending a cache.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215473590968418?view=List)
